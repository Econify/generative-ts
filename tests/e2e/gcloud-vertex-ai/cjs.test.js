const cjs = require("../../../packages/gcloud-vertex-ai/dist/index.cjs");

describe('@generative-ts/gcloud-vertex-ai cjs bundle', () => {
  test('api snapshot', async () => {
    expect(cjs).toMatchSnapshot();
  });

  test('sanity check', async () => {
    const model = await cjs.createVertexAiModelProvider({
      modelId: "gemini-1.0-pro",
      auth: {
        GCLOUD_LOCATION: 'us-central1',
        GCLOUD_PROJECT_ID: 'jmn-testing'
      }
    });

    const response = await model.sendRequest({
      $prompt: "Say hi:",
      max_tokens: 50,
      temperature: 0
    });

    expect(response.data.candidates).toBeDefined();
  });
});
