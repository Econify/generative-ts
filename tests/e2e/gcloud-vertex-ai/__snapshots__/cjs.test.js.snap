// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@generative-ts/gcloud-vertex-ai cjs bundle api snapshot 1`] = `
{
  "createVertexAiModelProvider": [Function],
}
`;
